src/Mod/Part/App/FeaturePartBoolean.cpp

```diff
@@ -39,6 +40,22 @@
 
 using namespace Part;
 
+void Part::throwIfInvalidIfCheckModel(const TopoDS_Shape& shape)
+{
+    Base::Reference<ParameterGrp> hGrp = App::GetApplication()

```

Comment:
```suggestion
    auto hGrp = App::GetApplication()
```

Is there a need to define the type explicitly?